go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -64,6 +66,249 @@ func NewVtctldServer(ts *topo.Server) *VtctldServer {
 	return &VtctldServer{ts: ts, tmc: tmclient.NewTabletManagerClient()}
 }
 
+// ChangeTabletType is part of the vtctlservicepb.VtctldServer interface.
+func (s *VtctldServer) ChangeTabletType(ctx context.Context, req *vtctldatapb.ChangeTabletTypeRequest) (*vtctldatapb.ChangeTabletTypeResponse, error) {
+	tablet, err := s.ts.GetTablet(ctx, req.TabletAlias)
+	if err != nil {
+		return nil, err
+	}
+
+	if !topo.IsTrivialTypeChange(tablet.Type, req.DbType) {
+		return nil, fmt.Errorf("tablet %v type change %v -> %v is not an allowed transition for ChangeTabletType", req.TabletAlias, tablet.Type, req.DbType)
+	}
+
+	if req.DryRun {
+		afterTablet := *tablet.Tablet
+		afterTablet.Type = req.DbType
+
+		return &vtctldatapb.ChangeTabletTypeResponse{
+			BeforeTablet: tablet.Tablet,
+			AfterTablet:  &afterTablet,
+			WasDryRun:    true,
+		}, nil
+	}
+
+	err = s.tmc.ChangeType(ctx, tablet.Tablet, req.DbType)
+	if err != nil {
+		return nil, err
+	}
+
+	changedTablet, _ := s.ts.GetTablet(ctx, req.TabletAlias)
```

Comment:
I'd at least catch the error and log ... something? getting a Nil back for after tablet on success is a bit weird and we'll want some kind trail if we hit it

Reply:
✅